pkg/graveler/staging/kv_migrate_test.go

```diff
@@ -0,0 +1,90 @@
+package staging_test
+
+import (
+	"bytes"
+	"context"
+	"math/rand"
+	"strconv"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/graveler"
+	"github.com/treeverse/lakefs/pkg/graveler/staging"
+	"github.com/treeverse/lakefs/pkg/kv"
+	kvparams "github.com/treeverse/lakefs/pkg/kv/params"
+	kvpg "github.com/treeverse/lakefs/pkg/kv/postgres"
+	"github.com/treeverse/lakefs/pkg/testutil"
+)
+
+type stagingTestRecord struct {
+	stagingToken graveler.StagingToken
+	records      []graveler.ValueRecord
+}
+
+func TestMigrate(t *testing.T) {
+	ctx := context.Background()
+	conn, _ := testutil.GetDB(t, databaseURI)
+	store, err := kv.Open(context.Background(), kvpg.DriverName, kvparams.KV{Postgres: &kvparams.Postgres{ConnectionString: databaseURI}})
+	testutil.MustDo(t, "Open KV Store", err)
+	kvStore := kv.StoreMessage{Store: store}
+	t.Cleanup(func() {
+		_, err = conn.Pool().Exec(ctx, `TRUNCATE kv`)
+		if err != nil {
+			t.Fatalf("failed to delete KV table from postgres DB %s", err)
+		}
+		store.Close()
+	})
+	dbMgr := staging.NewDBManager(conn)
+
+	data := createMigrateTestData(t, ctx, dbMgr, 300)
+
+	buf := bytes.Buffer{}
+	err = staging.Migrate(ctx, conn.Pool(), &buf)
+	require.NoError(t, err)
+
+	testutil.MustDo(t, "Import file", kv.Import(ctx, &buf, kvStore.Store))
+	kvMgr := staging.NewManager(kvStore)
+	for _, entry := range data {
+		for _, record := range entry.records {
+			r, err := kvMgr.Get(ctx, entry.stagingToken, record.Key)
+			require.NoError(t, err)
+			require.Equal(t, record.Value, r)
+		}
+	}
+}
+
+func genRandomBytes(t *testing.T, size int) (blk []byte) {
+	t.Helper()
+	blk = make([]byte, size)
+	_, err := rand.Read(blk)
+	testutil.MustDo(t, "Generate random bytes", err)
+	return
+}
+
+func createMigrateTestData(t *testing.T, ctx context.Context, mgr graveler.StagingManager, size int) []stagingTestRecord {
+	t.Helper()
+	d := make([]stagingTestRecord, 0)
+	for i := 0; i < size; i++ {
+		tokenData := stagingTestRecord{
+			stagingToken: graveler.StagingToken("test_token_" + strconv.Itoa(i)),
+			records:      make([]graveler.ValueRecord, 0),
+		}
+		for j := 0; j < rand.Intn(size); j++ {
```

Comment:
Trying to avoid more flakiness: Will a random 0 that creates no entries results with a failing test?

Reply:
The point here was to allow a test of a staging token with no records. Moreover - maybe we should explicitly have one with no records and not rely on randomness